### PR TITLE
Remove merge transition code

### PR DESCRIPTION
**Pull request by dapplion**
## Summary

Remove merge transition code that is no longer needed. The merge (Ethereum's transition from proof-of-work to proof-of-stake) occurred in September 2022. All related transition logic, readiness checks, and terminal difficulty validation can now be safely removed.

This PR represents the squashed and rebased version of the work at commit `ca6a1b66791e29b9ab47b6faa67424ede0281a06`.

## Changes

This PR removes approximately 2,100 net lines of code across 39 files, including:

- **Terminal difficulty detection**: Removed `get_terminal_pow_block_hash()`, `get_pow_block_at_total_difficulty()`, and related terminal block hash validation logic from execution layer
- **Bellatrix readiness checks**: Simplified `bellatrix_readiness.rs` from merge transition readiness validation to just genesis execution payload consistency checks  
- **Optimistic transition block verification**: Deleted entire `otb_verification_service.rs` (369 lines) and related optimistic import logic specific to the merge transition
- **Merge-specific notifications**: Removed terminal block found logging, merge readiness warnings, and transition status messages from notifier
- **Operation pool**: Cleaned up merge-specific attestation and block proposal logic that handled transition edge cases
- **Test infrastructure**: Removed merge transition test cases from `bellatrix.rs` (212 lines), `capella.rs` (156 lines), and other integration tests

## Files Changed

Key modifications:
- `beacon_node/beacon_chain/src/otb_verification_service.rs` - **deleted**
- `beacon_node/beacon_chain/src/bellatrix_readiness.rs` - simplified to genesis checks only
- `beacon_node/execution_layer/src/lib.rs` - removed terminal difficulty methods
- `beacon_node/client/src/notifier.rs` - removed merge readiness logging
- `beacon_node/operation_pool/src/lib.rs` - cleaned up transition-specific logic
- Various test files - removed obsolete merge transition tests

## Testing

- Existing test suite passes
- Removed tests were specific to merge transition scenarios that are no longer relevant
- Core functionality (block processing, attestations, execution layer integration) unchanged

### CODE REVIEW

**Comment by michaelsproul** [comment]
Have we tested syncing through the merge transition on e.g. mainnet?

**Comment by dapplion** [comment]
If we are okay dropping genesis sync, the next step is https://github.com/dapplion/lighthouse/pull/64

**Comment by chong-he** [comment]
Tested checkpoint-sync from `slot 4635712` = `epoch 144866` which is 30 epochs before the Bellatrix hard fork.

Most logs are OK, just repeatedly the following logs and it is able to sync pass the merge:
```
Feb 11 09:19:17.000 INFO  Syncing                                       peers: "194", distance: "8864079 slots (175 weeks 6 days)", speed: "6.67 slots/sec", est_time: "2 weeks 1 days"
Feb 11 09:19:17.000 INFO  Ready for fulu                                info: "ensure the execution endpoint is updated to the latest release"
```

Occasionally seeing this log:
```
Feb 11 08:45:37.567 WARN  Chain failed! Syncing to its head won't be retried for at least the next 30 seconds  id: 12316
```

but I can see the sync continues:

```
Feb 11 08:45:41.000 INFO  Syncing                                       peers: "88", distance: "8879015 slots (176 weeks 1 days)", speed: "5.33 slots/sec", est_time: "2 weeks 5 days"
Feb 11 08:45:53.000 INFO  Syncing                                       peers: "90", distance: "8878920 slots (176 weeks 1 days)", speed: "6.00 slots/sec", est_time: "2 weeks 3 days"
```

I think that's fine? 
Another note is peer count fluctuates, but I think that's not related to this change?

Nethermind is logging:
```
Pivot block from Consensus Layer isn't higher than pivot from initial config. New PivotBlockNumber: 15643562, old: 23456000
```
I think that's only because Lighthouse hasn't synced enough to the head yet

**Comment by dapplion** [comment]
@chong-he seeing "Ready for fulu" when syncing is not normal in previous LH versions?

**Comment by chong-he** [comment]
> @chong-he seeing "Ready for fulu" when syncing is not normal in previous LH versions?

From my understanding the log `Ready for Fulu` is normal in that case, because the node is still syncing (far back from the head)

File: beacon_node/http_api/tests/broadcast_validation_tests.rs [hunk 1]
```diff
@@ -1957,6 +1974,16 @@ pub async fn duplicate_block_status_code() {
...
     let num_initial: u64 = 31;
     let duplicate_block_status_code = StatusCode::IM_A_TEAPOT;
+
+    // First create a basic tester to check if deneb is enabled, which is required for blobs.
+    let tester_check = InteractiveTester::<E>::new(None, validator_count).await;
+    let state = tester_check.harness.get_current_state();
+    let fork_name = state.fork_name(&tester_check.harness.spec).unwrap();
+    if !fork_name.deneb_enabled() {
+        return;
+    }
```
Comment (michaelsproul) [inline beacon_node/http_api/tests/broadcast_validation_tests.rs]: This should just call `test_spec` instead of creating a whole-ass harness

File: beacon_node/beacon_chain/tests/block_verification.rs [hunk 2]
```diff
@@ -1543,157 +1542,6 @@ async fn verify_block_for_gossip_doppelganger_detection() {
     }
 }
 
-#[tokio::test]
-async fn add_base_block_to_altair_chain() {
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/tests/block_verification.rs:1547]: This test might still work? Maybe we should try restoring it

File: beacon_node/execution_layer/src/test_utils/mock_execution_layer.rs [hunk 3]
```diff
@@ -294,22 +283,6 @@ impl<E: EthSpec> MockExecutionLayer<E> {
...
-
-    pub fn move_to_terminal_block(self) -> Self {
-        self.server
-            .execution_block_generator()
-            .move_to_terminal_block()
-            .unwrap();
-        self
-    }
-
     pub fn produce_forked_pow_block(self) -> (Self, ExecutionBlockHash) {
```
Comment (michaelsproul) [inline beacon_node/execution_layer/src/test_utils/mock_execution_layer.rs]: Can probably delete this (should be unused now)

File: beacon_node/execution_layer/src/test_utils/mock_execution_layer.rs
Comment (michaelsproul) [inline beacon_node/execution_layer/src/test_utils/mock_execution_layer.rs]: Can probably delete this too.

File: beacon_node/execution_layer/src/test_utils/mod.rs [hunk 4]
```diff
@@ -138,22 +127,17 @@ impl<E: EthSpec> MockServer<E> {
...
-            terminal_difficulty,
-            terminal_block,
-            terminal_block_hash,
             server_config,
             shanghai_time,
             cancun_time,
             prague_time,
             osaka_time,
             amsterdam_time,
+            ..
```
Comment (michaelsproul) [inline beacon_node/execution_layer/src/test_utils/mod.rs]: Should probably remove this so we don't accidentally miss fields added in future

**Review by michaelsproul — APPROVED** [review]
Looks good! Keen to merge this soon so we can simplify our lives when writing Gloas tests :tada:
